File: Lab1.24ToDoFullStack/Lab1.24ToDoFullStack/Models/ToDoItem.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace Lab1._24ToDoFullStack.Models
{
    public class ToDoItem
    {
        [Key] //all matches the DB
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string AssignedTo { get; set; }
        public int Duration { get; set; }
        public bool IsCompleted { get; set; }

    }
}

File: Lab1.24ToDoFullStack/Lab1.24ToDoFullStack/ClientApp/src/app/ToDoItem.ts
export interface ToDoItem{
    id : number;
    name: string;
    description: string;
    assignedTo: string;
    duration: number;
    isCompleted: boolean;
}
export class Convert {
    public static toItem(json: string): ToDoItem {
        return JSON.parse(json);
    }
    public static toDoToJson(value: ToDoItem): string {
        return JSON.stringify(value);
    }
    public static toItemArray(json: string): ToDoItem[] {
        return JSON.parse(json);
    }
    public static toDoArrayToJson(value: ToDoItem[]): string {
        return JSON.stringify(value);
    }
}